File: app/src/main/java/de/bitb/astroskop/controller/BaseController.java
package de.bitb.astroskop.controller;

import javax.inject.Inject;

import de.bitb.astroskop.AstroApplication;
import de.bitb.astroskop.injection.IInjection;
import de.bitb.astroskop.model.Profile;
import de.bitb.astroskop.utils.SharedPreferencesUtils;
import lombok.Getter;

@Getter
public class BaseController {

    @Inject
    protected SharedPreferencesUtils sharedPreferencesUtils;

    public BaseController(AstroApplication astroApplication){
        if (this instanceof IInjection) {
            ((IInjection) this).inject(astroApplication.getAppComponent());
        } else {
            astroApplication.getAppComponent().inject(this);
        }
    }
 }

File: app/src/main/java/de/bitb/astroskop/controller/BaseController.kt
package de.bitb.astroskop.controller

import javax.inject.Inject

import de.bitb.astroskop.BaseApplication
import de.bitb.astroskop.BaseInjection
import de.bitb.astroskop.injection.IInjection
import de.bitb.astroskop.utils.SharedPreferencesUtils
import lombok.Getter

@Getter
open class BaseController(baseApplication: BaseApplication) : BaseInjection {

    @Inject
    var sharedPreferencesUtils: SharedPreferencesUtils? = null

    init {
        if (this is IInjection) {
            (this as IInjection).inject(baseApplication.appComponent)
        } else {
            baseApplication.appComponent!!.inject(this)
        }
    }
}
